### ISSUE TITLE
Django 4.0 collectstatic fails with 'js/load-image.all.min.js.map' could not be found

### ISSUE THREAD

**Issue by salomvary** [cloudinary/pycloudinary#305]
## Bug report for Cloudinary Python SDK
Before proceeding, please update to the latest version and test if the issue persists

## Describe the bug in a sentence or two.

When using Django 4 with `STATICFILES_STORAGE = "django.contrib.staticfiles.storage.ManifestStaticFilesStorage"` setting, running `./manage.py collectstatic` fails with the following exception:

`ValueError: The file 'js/load-image.all.min.js.map' could not be found with <django.contrib.staticfiles.storage.ManifestStaticFilesStorage object at 0x109d44190>.`

## Issue Type (Can be multiple)
- [ ] Build - Can’t install or import the SDK
- [ ] Performance - Performance issues
- [ ] Behaviour - Functions are not working as expected (such as generate URL)
- [ ] Documentation - Inconsistency between the docs and behaviour
- [x] Other (Specify): Failure when using Django static files

## Steps to reproduce

```
pip install django cloudinary
django-admin startproject cloudinary_bug
cd cloudinary_bug
# Replace cloudinary_bug/settings.py with settings.py below
./manage.py collectstatic --noinput
```

settings.py
```
"""
Django settings for cloudinary_bug project.

Generated by 'django-admin startproject' using Django 4.0.1.

For more information on this file, see
https://docs.djangoproject.com/en/4.0/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/4.0/ref/settings/
"""

from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.0/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-nffuht3hv*hx&q!t6b#xdo2$t2pzn$rq8d!)bru3mw!0v!%qgb'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = []


# Application definition

INSTALLED_APPS = [
    'cloudinary',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]

ROOT_URLCONF = 'cloudinary_bug.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]

WSGI_APPLICATION = 'cloudinary_bug.wsgi.application'


# Database
# https://docs.djangoproject.com/en/4.0/ref/settings/#databases

DATABASES = {
    'default': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': BASE_DIR / 'db.sqlite3',
    }
}


# Password validation
# https://docs.djangoproject.com/en/4.0/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/4.0/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.0/howto/static-files/

STATIC_URL = 'static/'

# Default primary key field type
# https://docs.djangoproject.com/en/4.0/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

STATIC_ROOT = "staticfiles"
STATICFILES_STORAGE = "django.contrib.staticfiles.storage.ManifestStaticFilesStorage"

```

## Error screenshots or Stack Trace (if applicable)

```
$ ./manage.py collectstatic --noinput
Post-processing 'js/load-image.all.min.js' failed!

Traceback (most recent call last):
  File "/private/tmp/cloudinary_bug/./manage.py", line 22, in <module>
    main()
  File "/private/tmp/cloudinary_bug/./manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/core/management/__init__.py", line 425, in execute_from_command_line
    utility.execute()
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/core/management/base.py", line 373, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/core/management/base.py", line 417, in execute
    output = self.handle(*args, **options)
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 187, in handle
    collected = self.collect()
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 134, in collect
    raise processed
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/contrib/staticfiles/storage.py", line 312, in _post_process
    content = pattern.sub(converter, content)
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/contrib/staticfiles/storage.py", line 198, in converter
    hashed_url = self._url(
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/contrib/staticfiles/storage.py", line 135, in _url
    hashed_name = hashed_name_func(*args)
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/contrib/staticfiles/storage.py", line 362, in _stored_name
    cache_name = self.clean_name(self.hashed_name(name))
  File "/Users/mrc/.pyenv/versions/3.10.0/envs/cloudinary-tmp/lib/python3.10/site-packages/django/contrib/staticfiles/storage.py", line 97, in hashed_name
    raise ValueError("The file '%s' could not be found with %r." % (filename, self))
ValueError: The file 'js/load-image.all.min.js.map' could not be found with <django.contrib.staticfiles.storage.ManifestStaticFilesStorage object at 0x10e7a8190>.
(cloudinary-tmp) 
```

## Operating System
- [ ] Linux
- [ ] Windows
- [x] macOS
- [ ] All

## Environment and Frameworks (fill in the version numbers)

- Cloudinary Python SDK version - 1.28.0
- Python Version - 3.10.0
- Framework (Django, Flask, etc) - 4.0.1

## Repository

n/a

**Comment by salomvary** [comment]
Workaround: create empty source map files in one of the apps that have static files, e.g. `mkdir -p my_app/static/js && touch my_app/static/js/load-image.all.min.js.map`.

**Comment by Reliann** [comment]
I too have this issue, The simplest solution is to downgrade django to 3.2.10 . I assume It's a compability issue with django 4.0.1?

**Comment by Vdeub-cloudinary** [comment]
@salomvary @Reliann thanks for your insights, I have shared those with our SDK team for further investigations. We will get back to you as soon as we have some details to share.

**Comment by patrick-tolosa** [comment]
Hey all, thanks for raising this issue.

This looks like a breaking change with how Django handles third party vendors and JS `sourcemap` files

https://code.djangoproject.com/ticket/33353?cversion=0&cnum_hist=4

I can only imagine this affects many other vendors.
The thread above suggests a workaround or two,.
We'll update this thread once we have more information from our side.

**Comment by const-cloudinary** [comment]
The issue is resolved in [Version 1.28.1](https://github.com/cloudinary/pycloudinary/releases/tag/1.28.1)
